Kernel/Bus/PCI/GenericInitializer.cpp

```diff
@@ -241,11 +240,25 @@ void initialize()
 
                 u32 interrupt_controller_phandle = MUST(map_stream.read_cell());
                 auto const* interrupt_controller = device_tree.phandle(interrupt_controller_phandle);
+#if ARCH(AARCH64)
+                // The interrupt controller handle contains two extra cells on aarch64, but they don't
+                // contain anything of interest.
+                MUST(map_stream.discard(sizeof(u32) * 2));
+#endif
                 VERIFY(interrupt_controller);
 
                 auto interrupt_cells = interrupt_controller->get_property("#interrupt-cells"sv)->as<u32>();
+#if ARCH(RISCV64)
                 VERIFY(interrupt_cells == 1 || interrupt_cells == 2);
                 u64 interrupt = MUST(map_stream.read_cells(interrupt_cells));
+#elif ARCH(AARCH64)
+                VERIFY(interrupt_cells == 3);
+                MUST(map_stream.discard(sizeof(u32))); // This is the IRQ type.
+                u64 interrupt = MUST(map_stream.read_cell());
+                MUST(map_stream.discard(sizeof(u32))); // This is the trigger type.
```

Comment:
I don't really like that we are hardcoding the interrupt descriptor format here. But I guess we already did that on RISC-V.
Maybe add a FIXME and/or check if the compatible of the interrupt parent is a GIC (see the ARMv8Timer code).

(We also probably add a check if the parent has the `interrupt-controller` property because we don't handle nested interrupt nexuses in that code.)

This code probably doesn't work because you need to add 32 to get the GIC interrupt number (Interrupts 32-1019 are SPIs).